Float.Core/UX/INavigationContext.cs

```diff
@@ -42,25 +43,55 @@ public interface INavigationContext
         /// <param name="animated">Whether to animate the transition.</param>
         void PushPage(Page page, bool animated = true);
 
+        /// <summary>
+        /// Pushs a page onto the existing navigation stack.
+        /// </summary>
+        /// <param name="page">The page.</param>
+        /// <param name="animated">Whether to animate the transition.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task PushPageAsync(Page page, bool animated = true);
+
         /// <summary>
         /// Pop the top-most page off the existing navigation stack.
         /// </summary>
         /// <param name="animated">Whether to animate the transition.</param>
         void PopPage(bool animated = true);
 
+        /// <summary>
+        /// Pop the top-most page off the existing navigation stack.
+        /// </summary>
+        /// <param name="animated">Whether to animate the transition.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task PopPageAsync(bool animated = true);
+
         /// <summary>
         /// Modally present the specified page in the current context.
         /// </summary>
         /// <param name="page">The page.</param>
         /// <param name="animated">Whether to animate the transition.</param>
         void PresentPage(Page page, bool animated = true);
 
+        /// <summary>
+        /// Modally present the specified page in the current context.
+        /// </summary>
+        /// <param name="page">The page.</param>
+        /// <param name="animated">Whether to animate the transition.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task PresentPageAsync(Page page, bool animated = true);
+
         /// <summary>
         /// Dismisses a modally-presented page.
         /// </summary>
         /// <param name="animated">Whether to animate the transition.</param>
         void DismissPage(bool animated = true);
 
+        /// <summary>
+        /// Dismisses a modally-presented page.
+        /// </summary>
+        /// <param name="animated">Whether to animate the transition.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task DismissPageAsync(bool animated = true);
```

Comment:
@EBusch Thoughts on updating the existing methods instead of adding new ones?
One the one hand, I would almost always say we should add to a library's API and avoid changing it to remain backward compatible.
On the other hand, in this case, this change would require implementations to make updates anyway (and thus isn't backwards compatible).

Reply:
@bbeversdorf thoughts on this? I think I lean towards the approach that doesn't break anything, though perhaps we consider marking these other methods as obsolete and remove them in a future update?